apps/mosquitto_passwd/mosquitto_passwd.c

```diff
@@ -629,13 +629,13 @@ int main(int argc, char *argv[])
 			return 1;
 		}
 
-		backup_file = malloc((size_t)strlen(password_file)+strlen(".backup.XXXXXX"));
+		backup_file = malloc((size_t)strlen(password_file)+strlen(".backup.XXXXXX")+1);
```

Comment:
Nit: pull the size out to a separate variable so you don't need to repeat the calculation below.